distributed/tests/test_cancelled_state.py

```diff
@@ -452,3 +454,113 @@ async def get_data(self, comm, *args, **kwargs):
     # w1 closed
 
     assert await f4 == 6
+
+
+@pytest.mark.parametrize("wait_for_processing", [True, False])
+@pytest.mark.parametrize("raise_error", [True, False])
+@gen_cluster(client=True, timeout=3)
+async def test_resumed_cancelled_handle_compute(
+    c, s, a, b, raise_error, wait_for_processing
+):
+    """
+    Given the history of a task
+    Executing -> Cancelled -> Fetch -> Resumed
+
+    A handle_compute should properly restore executing
+    """
+    # This test is heavily using set_restrictions to simulate certain scheduler
+    # decisions of placing keys
+
+    lock_compute = Lock()
+    await lock_compute.acquire()
+    enter_compute = Event()
+
+    def block(x, lock, enter_event):
+        enter_event.set()
+        with lock:
+            if raise_error:
+                raise RuntimeError("test error")
+            else:
+                return x + 1
+
+    f1 = c.submit(inc, 1, key="f1", workers=[a.address])
+    f2 = c.submit(inc, f1, key="f2", workers=[a.address])
+    f3 = c.submit(
+        block,
+        f2,
+        lock=lock_compute,
+        enter_event=enter_compute,
+        key="f3",
+        workers=[b.address],
+    )
+
+    f4 = c.submit(sum, [f1, f3], workers=[b.address])
+
+    await enter_compute.wait()
+
+    async def release_all_futures():
+        f1.release()
+        f2.release()
+        f3.release()
+        f4.release()
+
+        while any(fut.key in s.tasks for fut in [f1, f2, f3, f4]):
+            await asyncio.sleep(0.05)
+
+    await release_all_futures()
+    assert b.tasks[f3.key].state == "cancelled"
+
+    f1 = c.submit(inc, 1, key="f1", workers=[a.address])
+    f2 = c.submit(inc, f1, key="f2", workers=[a.address])
+    f3 = c.submit(
+        inc,
+        f2,
+        key="f3",
+        workers=[a.address],
```

Comment:
```suggestion
        workers=[a.address]
```
No need for multiple lines